[KOGITO-3039] - Agnostic infrastructure layer configuration on Operator (#2911)
kiegroup/kie-docs: doc-content/kogito-docs/src/main/asciidoc/openshift/con-kogito-operator-architecture.adoc

* *https://github.com/strimzi/strimzi-kafka-operator[Strimzi Operator]*: Used to deploy Apache Kafka clusters with Zookeeper for messaging in {PRODUCT} services
* *https://github.com/keycloak/keycloak-operator[Keycloak Operator]*: Used to deploy Keycloak server instances for security and single sign-on capabilities in {PRODUCT} services

When you enable an infrastructure mechanism by {PRODUCT} Infra deployment, the {PRODUCT} Operator uses the relevant third-party operator to create the infrastructure.

For example, the following `kogito-infinispan-infra` custom resource uses the `spec.resource.apiVersion` and `spec.resource.kind` configurations to enable Infinispan persistence for the {PRODUCT} service:

.Example {PRODUCT} infra resource with persistence
[source,yaml]
----
apiVersion: app.kiegroup.org/v1alpha1
kind: KogitoInfra
metadata:
  name: kogito-infinispan-infra
spec:
  resource:
    apiVersion: infinispan.org/v1
    kind: Infinispan
----

> In this example, if infinispan instance with name `kogito-infinispan` not exits then {PRODUCT} Operator uses the Infinispan Operator to deploy the Infinispan Server instance for persistence.

Reviewer: ```suggestion
In this example, if infinispan instance with name `kogito-infinispan` does not exist then {PRODUCT} Operator uses the Infinispan Operator to deploy the Infinispan Server instance for persistence.
```

Author: done

> Similarly for example, the following `kogito-kafka-infra` custom resource uses the `spec.resource.apiVersion` and `spec.resource.kind` configurations to enable Kafka messaging for the {PRODUCT} service:

Reviewer: ```suggestion
Similarly to the previous example, the following `kogito-kafka-infra` custom resource uses the `spec.resource.apiVersion` and `spec.resource.kind` configurations to enable Kafka messaging for the {PRODUCT} service:
```

Author: done

.Example {PRODUCT} infra resource with messaging
[source,yaml]
----
apiVersion: app.kiegroup.org/v1alpha1
kind: KogitoInfra
metadata:
  name: kogito-kafka-infra
spec:
  resource:
    apiVersion: kafka.strimzi.io/v1beta1
    kind: Kafka
----

> In this example, if kafka instance with name `kogito-kafka` not exits then {PRODUCT} Operator uses the Strimzi Operator to deploy the Kafka cluster for event messaging.

Reviewer: ```suggestion
In this example, if kafka instance with name `kogito-kafka` does not exist then {PRODUCT} Operator uses the Strimzi Operator to deploy the Kafka cluster for event messaging.
```

Author: done

If you have your custom infra resource then you could provide its reference through `spec.resource.name` and `spec.resource.namespace` configurations.
.Example {PRODUCT} infra resource with custom messaging resource
[source,yaml]
----
apiVersion: app.kiegroup.org/v1alpha1
kind: KogitoInfra
metadata:
  name: my-kafka-infra
spec:
  resource:
    apiVersion: kafka.strimzi.io/v1beta1
    kind: Kafka
    name: my-kafka-instance
    namespace: my-namespace
----

> In this example, `KogitoInfra` does not deploy Kafka cluster, but connect to the kafka instance with name `my-kafka` for event messaging.

Reviewer: ```suggestion
In this example, `KogitoInfra` does not deploy Kafka cluster, but connect to the kafka instance with name `my-kafka` from the namespace `my-namespace` for event messaging.
```

Author: done

  name: data-index
spec:
  replicas: 1
  infra:
     - kogito-infinispan-infra
     - kogito-kafka-infra
----

> You should create `kogito-infinispan-infra` and `kogito-kafka-infra` prior to setup `KogitoDataIndex`

Reviewer: ```suggestion
You should create `kogito-infinispan-infra` and `kogito-kafka-infra` prior to the setup of `KogitoDataIndex`.
```

Author: done
